src/main/java/de/dentrassi/rpm/builder/RpmMojo.java

```diff
@@ -1323,7 +1342,14 @@ private RpmVersion makeVersion() {
             this.logger.info("Building with SNAPSHOT version from project: %s", baseVersion);
             return new RpmVersion(this.epoch, baseVersion, makeSnapshotReleaseString());
         }
-        return new RpmVersion(this.epoch, this.version, this.release);
+        String version = this.version;
+        if (modifyVersion) {
+            for (VersionSubstitution versionSubstitution : versionReplacements) {
+                logger.info("For version %s looking for %s to replace with %s", version, versionSubstitution.search, versionSubstitution.replace);
+                version = version.replaceAll(versionSubstitution.search, versionSubstitution.replace);
+            }
+        }
+        return new RpmVersion(this.epoch, version, this.release);
```

Comment:
If the version is invalid, it could eventually be detected here. Technically, the output filename doesn't have to be a valid `RpmVersion`, which means we would have to change `RpmFileNameProvider` to use `String` instead of `RpmVersion`, or it would fail at that point with using the non-transformed version (since `makeVersion` called with `false` there).

Reply:
I didn't want to change the output file name as that is what is used in maven storage (and is legal). I'm guessing you're referring to https://github.com/eclipse/packager/pull/76 which would break rpm-builder here if an invalid version is used ; I'm beginning to think the validation might actually cause more problems that it solves.